src/main/java/com/iexec/commons/poco/chain/SignerService.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2020-2024 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.commons.poco.chain;
+
+import com.iexec.commons.poco.eip712.EIP712Entity;
+import com.iexec.commons.poco.security.Signature;
+import com.iexec.commons.poco.utils.EthAddress;
+import com.iexec.commons.poco.utils.SignatureUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.web3j.crypto.*;
+import org.web3j.crypto.exception.CipherException;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.Response;
+import org.web3j.protocol.core.methods.response.EthSendTransaction;
+import org.web3j.protocol.core.methods.response.Transaction;
+import org.web3j.protocol.exceptions.JsonRpcError;
+import org.web3j.tx.RawTransactionManager;
+import org.web3j.utils.Numeric;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+
+@Slf4j
+public class SignerService {
+
+    private final Credentials credentials;
+    private final RawTransactionManager txManager;
+    private final Web3j web3j;
+
+    public SignerService(Web3j web3j, long chainId) throws Exception {
+        try {
+            ECKeyPair ecKeyPair = Keys.createEcKeyPair();
+            credentials = Credentials.create(ecKeyPair);
+            log.info("Created new wallet credentials [address:{}] ", credentials.getAddress());
+        } catch (NoSuchAlgorithmException | NoSuchProviderException | InvalidAlgorithmParameterException e) {
+            log.error("Cannot create new wallet credentials", e);
+            throw e;
+        }
+        this.txManager = new RawTransactionManager(
+                web3j, credentials, chainId, 10, 1000L);
+        this.web3j = web3j;
+    }
+
+    public SignerService(Web3j web3j, long chainId, Credentials credentials) {
+        this.credentials = credentials;
```

Comment:
it's not essential, but we assign before doing the test. I know it's already like that in CredentialsAbstractService

Reply:
please see 0cdb55